src/Config.cpp

```diff
@@ -86,12 +178,15 @@ void Config::readFromMap(map<string, any> &mp) {
     for (auto kv : mp) {
         auto key = kv.first;
         if (node_[key]) {
-            if (strcmp(kv.second.type().name(), "i") == 0) {
-                node_[key] = any_cast<int>(kv.second);
-            } else if (strcmp(kv.second.type().name(), "b") == 0) {
-                node_[key] = any_cast<bool>(kv.second);
-            } else if (strcmp(kv.second.type().name(), "c") == 0) {
-                node_[key] = any_cast<char>(kv.second);
+            if(datatype[key]== 'i'){
```

Comment:
A `switch` statement here makes the code more readable.
Also don't forget to use`clang-format`